src/main/java/org/mitre/synthea/export/rif/RIFExporter.java

```diff
@@ -0,0 +1,267 @@
+package org.mitre.synthea.export.rif;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import org.mitre.synthea.helpers.Config;
+import org.mitre.synthea.world.agents.Person;
+import org.mitre.synthea.world.agents.Provider;
+import org.mitre.synthea.world.concepts.HealthRecord;
+
+/**
+ * Base class for all of the type-specific exporters.
+ */
+public abstract class RIFExporter {
+
+  protected static final String BB2_PARTD_CONTRACTS = "BB2_PARTD_CONTRACTS";
+  protected static final AtomicLong nextCarrClmCntlNum = new AtomicLong(Config.getAsLong(
+          "exporter.bfd.carr_clm_cntl_num_start", -1));
+  protected static final AtomicLong nextPdeId = new AtomicLong(Config.getAsLong(
+          "exporter.bfd.pde_id_start", -1));
```

Comment:
Seems weird that some of these are in here... might make more sense to put things like `nextCarrClmCntlNum` or `nextPdeId` (possibly others) in the specific exporter classes that inherit from this base class.

Reply:
I've moved the ids that are specific to a particular file to the corresponding exporter. General ids that are used by multiple exporters I've left in the base class.